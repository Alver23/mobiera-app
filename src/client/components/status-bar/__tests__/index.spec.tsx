// Dependencies
import React from 'react';
import * as RNModules from 'react-native';
import { shallow, ShallowWrapper } from 'enzyme';

// Under test file
import StatusBarComponent from '../index';

describe('<StatusBar />', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<StatusBarComponent />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(component.find(RNModules.StatusBar).exists()).toBeTruthy();
  });

  it('should render correctly when dark mode is true', () => {
    jest.spyOn(RNModules, 'useColorScheme').mockReturnValue('dark');
    component.setProps({});
    const barStyle = component.find(RNModules.StatusBar).prop('barStyle');
    expect(barStyle).toBe('light-content');
  });
});
