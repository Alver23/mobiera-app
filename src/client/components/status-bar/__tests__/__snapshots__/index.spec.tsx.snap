// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBar /> should save a snapshot of the component 1`] = `
<StatusBar
  animated={false}
  barStyle="dark-content"
  showHideTransition="fade"
/>
`;
